### PR TITLE
Feegrind fixes and test

### PR DESCRIPTION
**Pull request by rustyrussell**

### CODE REVIEW

File: onchaind/onchain.c [hunk 1]
```diff
@@ -97,99 +100,105 @@ struct tracked_output {
...
+			     const u8 *wscript,
+			     u64 multiplier)
 {
 	u64 prev_fee = UINT64_MAX;
-	u64 input_amount = *commit_tx->input[0].amount;
+	u64 input_amount = *tx->input[0].amount;
 
-	for (s64 i = feerate_range.max; i >= feerate_range.min; i--) {
+	for (u64 i = min_possible_feerate; i <= max_possible_feerate; i++) {
 		u64 fee = i * multiplier / 1000;
```
Comment (ZmnSCPxj) [inline onchaind/onchain.c]: I remember branch `bitcoind-vbytes-must-die-die-die` needed to add some complicated computations to match bitcoind computation of feerate, should we also use it here?  Or this is sufficient? How does e.g. lnd compute this?

**Comment by ZmnSCPxj** [comment]
ACK de4f2c29edc826c3af22a4f39bbf0d55b53eebab

modulo question on whether we should also be using the bitcoind-based feerate or not.

**Comment by cdecker** [comment]
Concept ACK

CI complains about:
 - Missing DEVELOPER=1 guard on the new test
 - Whitespace in blank lines in the python test
 - valgrind error in run-grind_feerate.c

```
==176== 208 bytes in 1 blocks are still reachable in loss record 1 of 2
==176==    at 0x4C2DB8F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==176==    by 0x43567F: checked_malloc (util.h:72)
==176==    by 0x43567F: secp256k1_context_create (secp256k1.c:69)
==176==    by 0x425578: main (run-grind_feerate.c:162)
==176== 
==176== 1,048,576 bytes in 1 blocks are still reachable in loss record 2 of 2
==176==    at 0x4C2DB8F: malloc (in /usr/lib/valgrind/vgpreload_memcheck-amd64-linux.so)
==176==    by 0x435825: checked_malloc (util.h:72)
==176==    by 0x435825: secp256k1_ecmult_context_build (ecmult_impl.h:174)
==176==    by 0x435825: secp256k1_context_create (secp256k1.c:87)
==176==    by 0x425578: main (run-grind_feerate.c:162)
==176==
```

**Comment by rustyrussell** [comment]
Rebased, ran update-mocks to fix missing stub.

**Pull request merged**